### ISSUE TITLE
Persist Charge (3/4) - Cancel stop flow

### ISSUE THREAD

**Issue by prtandrup** [Energinet-DataHub/geh-charges#1152]
**Synopsis:**
As a Grid Access Provider
I want to cancel a stop of charge
So that I may correct a stop mistake

This is another story of a few stories with the goal of implementing the [charge persistence rule set](https://github.com/Energinet-DataHub/geh-charges/tree/main/docs/process-flows#persist-charge) (master data only)

With this story we will implement the **cancel stop** flow, highlighted below:
<hr>

![image.png](https://images.zenhubusercontent.com/60ae42d0916bb92d8b000901/c65a0358-6043-4c73-8419-e3f6f80517a1)

**Acceptance criteria:**
- [x] 1 Given an existing charge is stopped
When a charge request is received with an effective date (start) equal to the stop date
(And the time limit for doing this update is obeyed)
Then the stop is cancelled
And the persisted result must be like:

![image.png](https://images.zenhubusercontent.com/60ae42d0916bb92d8b000901/47bce285-e1aa-4b6a-b40a-52e443d83f8f)

**Comment by prtandrup** [comment]
A PR is on the way, which will remove the "stop charge not implemented" validation rule, which will unblock this story. 
This PR is related to story #1153

**Comment by prtandrup** [comment]
[PR](https://github.com/Energinet-DataHub/geh-charges/pull/1212) has been merged and deployed. Blocked label removed. Irene informed.

**Comment by IreneDataHub** [comment]
Test #1152 Cancel stop flow = OK

Create tarif D03stop2 start 1 Mai
Stop tarif D03stop2 1. July
Cancel stop 1. July for tariff D03stop2 enddatetime expect to be "9999-12-31T23:59:59" again

Create: 
                "ChargeId": "1aa1b228-2587-4155-8857-2e6ee5b68645",
                "TransparentInvoicing": false,
                "Description": "The charge description stop",
                "Name": "demo stop 2 test",
                "VatClassification": 1,
                "StartDateTime": "2022-04-30T22:00:00",
                "EndDateTime": "9999-12-31T23:59:59"
Stop
       "Description": "The charge description stop",
                "Name": "demo stop 2 test",
                "VatClassification": 1,
                "StartDateTime": "2022-04-30T22:00:00",
                "EndDateTime": "2022-06-30T22:00:00"

Start 1.7 cancel the stop

<?xml version="1.0" encoding="UTF-8"?>
<cim:RequestChangeOfPriceList_MarketDocument xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:cim="urn:ediel.org:structure:requestchangeofpricelist:0:1" xsi:schemaLocation="urn:ediel.org:structure:requestchangeofpricelist:0:1 urn-ediel-org-structure-requestchangeofpricelist-0-1.xsd">
	<cim:mRID>DocId{{$isoTimestamp}}</cim:mRID>
	<cim:type>D10</cim:type>
	<cim:process.processType>D18</cim:process.processType>
	<cim:businessSector.type>23</cim:businessSector.type>
	<cim:sender_MarketParticipant.mRID codingScheme="A10">{{GridAccessProvider}}</cim:sender_MarketParticipant.mRID>
	<cim:sender_MarketParticipant.marketRole.type>DDM</cim:sender_MarketParticipant.marketRole.type>
	<cim:receiver_MarketParticipant.mRID codingScheme="A10">5790001330552</cim:receiver_MarketParticipant.mRID>
	<cim:receiver_MarketParticipant.marketRole.type>DDZ</cim:receiver_MarketParticipant.marketRole.type>
	<cim:createdDateTime>{{ISO8601Timestamp}}</cim:createdDateTime>
	<cim:MktActivityRecord>
		<cim:mRID>Op1Id{{$isoTimestamp}}</cim:mRID>
		<cim:ChargeGroup>
			<cim:ChargeType>
				<cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">{{GridAccessProvider}}</cim:chargeTypeOwner_MarketParticipant.mRID>
				<cim:type>D03</cim:type>
				<cim:mRID>D03stop2</cim:mRID>
				<cim:name>demo stop 2 test</cim:name>
				<cim:description>The charge description stop</cim:description>
				<cim:priceTimeFrame_Period.resolution>P1D</cim:priceTimeFrame_Period.resolution>
				<cim:effectiveDate>2022-06-30T22:00:00Z</cim:effectiveDate>                           
				<cim:VATPayer>D01</cim:VATPayer>
				<cim:transparentInvoicing>false</cim:transparentInvoicing>
				<cim:taxIndicator>false</cim:taxIndicator>
				<cim:Series_Period>
					<cim:resolution>P1D</cim:resolution>
					<cim:timeInterval>
						<cim:start>2022-10-31T23:00Z</cim:start>
						<cim:end>2022-11-01T23:00Z</cim:end>
					</cim:timeInterval>
					<cim:Point>
						<cim:position>1</cim:position>
						<cim:price.amount>150.001</cim:price.amount>
					</cim:Point>
				</cim:Series_Period>
			</cim:ChargeType>
		</cim:ChargeGroup>
	</cim:MktActivityRecord>
	<cim:MktActivityRecord>
		<cim:mRID>Op2Id{{$isoTimestamp}}</cim:mRID>
		<cim:ChargeGroup>
			<cim:ChargeType>
				<cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">{{GridAccessProvider}}</cim:chargeTypeOwner_MarketParticipant.mRID>
				<cim:type>D03</cim:type>
				<cim:mRID>ChId{{$randomInt}}</cim:mRID>
				<cim:name>Tariff2 of Bundle</cim:name>
				<cim:description>The charge description</cim:description>
				<cim:priceTimeFrame_Period.resolution>P1D</cim:priceTimeFrame_Period.resolution>
				<cim:effectiveDate>2022-10-31T23:00:00Z</cim:effectiveDate>
				<cim:VATPayer>D01</cim:VATPayer>
				<cim:transparentInvoicing>false</cim:transparentInvoicing>
				<cim:taxIndicator>false</cim:taxIndicator>
				<cim:Series_Period>
					<cim:resolution>P1D</cim:resolution>
					<cim:timeInterval>
						<cim:start>2022-10-31T23:00Z</cim:start>
						<cim:end>2022-11-01T23:00Z</cim:end>
					</cim:timeInterval>
					<cim:Point>
						<cim:position>1</cim:position>
						<cim:price.amount>150.001</cim:price.amount>
					</cim:Point>
				</cim:Series_Period>
			</cim:ChargeType>
		</cim:ChargeGroup>
	</cim:MktActivityRecord>
</cim:RequestChangeOfPriceList_MarketDocument>

                "Id": "df4d02bb-5c80-4085-9a5b-19375d62db9f",
                "ChargeId": "1aa1b228-2587-4155-8857-2e6ee5b68645",
                "TransparentInvoicing": false,
                "Description": "The charge description stop",
                "Name": "demo stop 2 test",
                "VatClassification": 1,
                "StartDateTime": "2022-04-30T22:00:00",
                "EndDateTime": "9999-12-31T23:59:59"
            }
        ],
        "Id": "1aa1b228-2587-4155-8857-2e6ee5b68645",
        "SenderProvidedChargeId": "D03stop2",
        "Type": 3,
        "OwnerId": "0e223e42-bed4-4778-a973-8d0ad9813f71",
        "TaxIndicator": false,
        "Resolution": 3

**Comment by prtandrup** [comment]
### AC# 1 tested in T-001 ✅

**Starting point:**
Charge11 started and stopped:

![image.png](https://images.zenhubusercontent.com/60ae42d0916bb92d8b000901/55c16b89-62c6-4364-8e33-e1c6bdaaa1cb)

**Submitted cancel message with effective date (2022-05-05T22:00:00Z)**

````
<?xml version="1.0" encoding="UTF-8"?>
<cim:RequestChangeOfPriceList_MarketDocument xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:cim="urn:ediel.org:structure:requestchangeofpricelist:0:1" xsi:schemaLocation="urn:ediel.org:structure:requestchangeofpricelist:0:1 urn-ediel-org-structure-requestchangeofpricelist-0-1.xsd">
    <cim:mRID>DocId2022-03-24T13:06:40.533Z</cim:mRID>
    <cim:type>D10</cim:type>
    <cim:process.processType>D18</cim:process.processType>
    <cim:businessSector.type>23</cim:businessSector.type>
    <cim:sender_MarketParticipant.mRID codingScheme="A10">8100000000207</cim:sender_MarketParticipant.mRID>
    <cim:sender_MarketParticipant.marketRole.type>DDM</cim:sender_MarketParticipant.marketRole.type>
    <cim:receiver_MarketParticipant.mRID codingScheme="A10">5790001330552</cim:receiver_MarketParticipant.mRID>
    <cim:receiver_MarketParticipant.marketRole.type>DDZ</cim:receiver_MarketParticipant.marketRole.type>
    <cim:createdDateTime>2022-03-24T14:06:40Z</cim:createdDateTime>
    <cim:MktActivityRecord>
        <cim:mRID>OpId2022-03-24T13:06:40.533Z</cim:mRID>
        <cim:ChargeGroup>
            <cim:ChargeType>
                <cim:chargeTypeOwner_MarketParticipant.mRID codingScheme="A10">8100000000207</cim:chargeTypeOwner_MarketParticipant.mRID>
                <cim:type>D03</cim:type>
                <cim:mRID>Charge11</cim:mRID>
                <cim:name>Charge Tariff day Name2</cim:name>
                <cim:description>The charge description update</cim:description>
                <cim:priceTimeFrame_Period.resolution>P1D</cim:priceTimeFrame_Period.resolution>
                <cim:effectiveDate>2022-05-05T22:00:00Z</cim:effectiveDate>
                <!--cim:terminationDate>2022-05-05T22:00:00Z</cim:terminationDate-->
                <cim:VATPayer>D02</cim:VATPayer>
                <cim:transparentInvoicing>false</cim:transparentInvoicing>
                <cim:taxIndicator>true</cim:taxIndicator>
                <cim:Series_Period>
                    <cim:resolution>P1D</cim:resolution>
                    <cim:timeInterval>
                        <cim:start>2022-05-05T22:00Z</cim:start>
                        <cim:end>2022-05-06T22:00Z</cim:end>
                    </cim:timeInterval>
                    <cim:Point>
                        <cim:position>1</cim:position>
                        <cim:price.amount>200.123</cim:price.amount>
                    </cim:Point>
                </cim:Series_Period>
            </cim:ChargeType>
        </cim:ChargeGroup>
    </cim:MktActivityRecord>
</cim:RequestChangeOfPriceList_MarketDocument>
````

**Result:**

![image.png](https://images.zenhubusercontent.com/60ae42d0916bb92d8b000901/be9437e5-7792-49f0-9e8f-0c05f682bccd)
